tests/testthat/test-ggplot-coord.R

```diff
@@ -0,0 +1,53 @@
+context("Fixed coordinates")
+
+# Expect trace function
+expect_traces <- function(gg, n_traces, name) {
+  stopifnot(is.ggplot(gg))
+  stopifnot(is.numeric(n_traces))
+  save_outputs(gg, paste0("coord_fixed-", name))
+  L <- gg2list(gg)
+  all_traces <- L$data
+  no_data <- sapply(all_traces, function(tr) {
+    is.null(tr[["x"]]) && is.null(tr[["y"]])
+  })
+  has_data <- all_traces[!no_data]
+  expect_equal(length(has_data), n_traces)
+  list(traces = has_data, layout = L$layout)
+}
+
+# Data where x ranges from 0-10, y ranges from 0-30
+set.seed(202)
+dat <- data.frame(xval = runif(40,0,10), yval = runif(40,0,30))
+
+# Force equal scaling
+p <- ggplot(dat, aes(xval, yval)) + geom_point() + coord_fixed()
+# Test 
+test_that("coord_fixed() is translated to the right height-width ratio", {
+  info <- expect_traces(p, 1, "force_equal_scaling")
+  tr <- info$traces[[1]]
+  la <- info$layout
+  expect_identical(tr$type, "scatter")
+  # height-width ratio check
+  built <- ggplot_build2(p)
+  x_range <- range(built[[2]]$ranges[[1]]$x.major_source, na.rm = TRUE)
+  y_range <- range(built[[2]]$ranges[[1]]$y.major_source, na.rm = TRUE)
```

Comment:
That's true, but you aren't setting `xlim()` or `ylim()` in the test itself, so you currently aren't testing for that...

In general, using the *raw* data as ground truth in tests is a good idea. Suppose ggplot2 internals change at some point and the x range is no longer stored in this location: `built[[2]]$ranges[[1]]$x.major_source`. This test wouldn't necessarily detect any problems since you're basically testing that we map the ggplot object in a certain way (which may be wrong).